third_party/terraform/tests/resource_dataproc_cluster_test.go.erb

```diff
@@ -420,6 +420,37 @@ func TestAccDataprocCluster_withStagingBucket(t *testing.T) {
 	})
 }
 
+func TestAccDataprocCluster_withTempBucket(t *testing.T) {
+	t.Parallel()
+
+	rnd := randString(t, 10)
+	var cluster dataproc.Cluster
+	clusterName := fmt.Sprintf("tf-test-dproc-%s", rnd)
+	bucketName := fmt.Sprintf("%s-temp-bucket", clusterName)
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDataprocClusterDestroy(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataprocCluster_withTempBucketAndCluster(clusterName, bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDataprocClusterExists(t, "google_dataproc_cluster.with_bucket", &cluster),
+					resource.TestCheckResourceAttr("google_dataproc_cluster.with_bucket", "cluster_config.0.temp_bucket", bucketName)),
+			},
+			{
+				// Simulate destroy of cluster by removing it from definition,
+				// but leaving the temp bucket (should not be auto deleted)
+				Config: testAccDataprocCluster_withTempBucketOnly(bucketName),
```

Comment:
I'm not really sure why this step is here. It seems to be testing that deleting the cluster does not delete the storage bucket, but I'm not sure why we would be concerned about that. Declaring the storage bucket in a TF config and ensuring that it exists should be covered by storage bucket tests

Reply:
@slevenick I believe the contributor copy-pasted this from the storage bucket tests directly above. My assumption was that there was a concern that GCP might auto-delete the related storage (or temp) bucket resource without terraform being able to control it.

If you think this is unnecessary, I can definitely ask them to remove it & open a ticket to update the storage_bucket tests as well.

Reply:
talked with Sam - it's okay to leave this in place since it is the same as the other test.